packages/resolve/test/request-resolver.spec.ts

```diff
@@ -0,0 +1,344 @@
+            expect(resolveRequest('/node_modules/express', 'lodash/v2-specific-file'))
+                .to.be.resolvedTo('/node_modules/lodash/v2-specific-file.js')
+        })
+
+        it('resolves requests to scoped packages', () => {
+            const fs = createMemoryFs({
+                node_modules: {
+                    '@stylable': {
+                        cli: {
+                            'index.js': EMPTY,
+                            'test-utils.js': EMPTY
+                        }
+                    }
+                }
+            })
+            const resolveRequest = createRequestResolver({ fs })
+
+            expect(resolveRequest('/', '@stylable/cli'))
+                .to.be.resolvedTo('/node_modules/@stylable/cli/index.js')
+
+            expect(resolveRequest('/', '@stylable/cli/test-utils'))
+                .to.be.resolvedTo('/node_modules/@stylable/cli/test-utils.js')
+        })
+
+        it('allows specifying custom packages roots', () => {
+            const fs = createMemoryFs({
+                project: {
+                    third_party: {
+                        koa: {
+                            'package.json': '{"main": "main-index"}',
+                            'main-index.js': EMPTY
+                        }
+                    }
+                },
+                root_libs: {
+                    react: {
+                        'index.js': EMPTY
+                    }
+                }
+            })
+            const resolveRequest = createRequestResolver({ fs, packageRoots: ['third_party', '/root_libs'] })
+
+            expect(resolveRequest('/project', 'koa'))
+                .to.be.resolvedTo('/project/third_party/koa/main-index.js')
+
+            expect(resolveRequest('/project', 'react'))
+                .to.be.resolvedTo('/root_libs/react/index.js')
+        })
+    })
+
+    describe('browser field', () => {
+        it('prefers "browser" over "main" when loading a package.json', () => {
+            const fs = createMemoryFs({
+                lodash: {
+                    'package.json': '{"main": "entry.js", "browser": "./browser.js"}',
+                    'entry.js': EMPTY,
+                    'browser.js': EMPTY
+                }
+            })
+            const resolveRequest = createRequestResolver({ fs })
+
+            expect(resolveRequest('/', './lodash')).to.be.resolvedTo('/lodash/browser.js')
+        })
+
+        it('uses "browser" even if "main" was not defined', () => {
+            const fs = createMemoryFs({
+                lodash: {
+                    'package.json': '{"browser": "file.js"}',
+                    'file.js': EMPTY
+                }
+            })
+            const resolveRequest = createRequestResolver({ fs })
+
+            expect(resolveRequest('/', './lodash')).to.be.resolvedTo('/lodash/file.js')
+        })
+
+        it('prefers "main" when resolution "target" is set to "node"', () => {
```

Comment:
Should there be a complementing test for this checking `target: 'browser'`?

Reply:
added test, even though it's the default behavior and tested by first test as well. new test checks what happens if the user explicitly provided flag.